qst/src/main/java/io/deephaven/qst/array/Array.java

```diff
@@ -0,0 +1,60 @@
+package io.deephaven.qst.array;
+
+import io.deephaven.qst.type.Type;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.stream.Stream;
+
+/**
+ * Provides strong typing around an array-like object.
+ *
+ * @param <T> the generic type
+ * @see PrimitiveArray
+ * @see GenericArray
+ */
+public interface Array<T> extends Iterable<T>, Serializable {
+
+    static <T> ArrayBuilder<T, ?, ?> builder(Type<T> type) {
+        return TypeToArrayBuilder.of(type, 16);
+    }
+
+    static <T> ArrayBuilder<T, ?, ?> builder(Type<T> type, int initialCapacity) {
+        return TypeToArrayBuilder.of(type, initialCapacity);
+    }
+
+    static <T> Array<T> empty(Type<T> type) {
+        return builder(type, 0).build();
+    }
+
+    static <T> Array<T> of(Type<T> type, T... data) {
+        return builder(type, data.length).add(data).build();
+    }
+
+    static <T> Array<T> of(Type<T> type, Iterable<T> data) {
+        if (data instanceof Collection) {
```

Comment:
I've seen other libraries do it. It's sometimes useful if you are getting data from an interface where that interface returns Iterable instead of Collection:

`Array.of(type, someInterface.getData())`

That said, it's probably not worth it. Will remove for now.

Reply:
I re-added this back.